sdk/src/main/java/com/hedera/hashgraph/sdk/Client.java

```diff
@@ -404,6 +401,25 @@ private void cancelScheduledNetworkUpdate() {
         }
     }
 
+    private void cancelAllSubscriptions() {
+        ArrayList<SubscriptionHandle> handles = new ArrayList<>(subscriptions);
+        for (SubscriptionHandle handle : handles) {
+            handle.unsubscribe();
```

Comment:
Should enhance `SubscriptionHandle::unsubscribe` to set `onUnsubscribe = null` after `run()` so it only runs once even if invoked by user and Client